src/herder/HerderImpl.cpp

```diff
@@ -509,18 +509,46 @@ HerderImpl::processSCPQueue()
 void
 HerderImpl::processSCPQueueUpToIndex(uint64 slotIndex)
 {
-    while (true)
+    try
     {
-        SCPEnvelope env;
-        if (mPendingEnvelopes.pop(slotIndex, env))
-        {
-            getSCP().receiveEnvelope(env);
-        }
-        else
+        uint64 minSlotIndex = 0;
+        bool notTracking = !mHerderSCPDriver.trackingSCP();
+        while (true)
         {
-            return;
+            SCPEnvelope env;
+            if (mPendingEnvelopes.pop(minSlotIndex, slotIndex, env))
+            {
+                // ensures that regardless of what receiveEnvelope does, we do
+                // not process new messages the next time we loop if this would
+                // violate "tracking"'s invariant: in that we only process the
+                // current slot when tracking
```

Comment:
Rather than say "the current slot", I would find it easier to understand here if you referred to a specific notion in the herder, eg. its `nextConsensusLedgerIndex()` or such (which is what I think you mean, but I'm not sure).

Reply:
yeah this is actually not important, I reworked the comment